Source/Flow/Private/Nodes/FlowNodeBase.cpp

```diff
@@ -715,3 +743,285 @@ bool UFlowNodeBase::BuildMessage(FString& Message) const
 	return false;
 }
 #endif
+
+EFlowDataPinResolveResult UFlowNodeBase::TryResolveDataPinPrerequisites(const FName& PinName, const UFlowNode*& FlowNode, const FFlowPin*& FlowPin, EFlowPinType PinType) const
+{
+	FlowNode = GetFlowNodeSelfOrOwner();
+
+	if (!IsValid(FlowNode))
+	{
+		LogError(FString::Printf(TEXT("Unexpected for %s to not have an associated FlowNode"), *GetName()), EFlowOnScreenMessageType::Temporary);
+
+		return EFlowDataPinResolveResult::FailedWithError;
+	}
+
+	FlowPin = FindFlowPinByName(PinName, FlowNode->GetInputPins());
+	if (!FlowPin)
+	{
+		LogError(FString::Printf(TEXT("Unexpected to find pin named %s"), *PinName.ToString()), EFlowOnScreenMessageType::Temporary);
+
+		return EFlowDataPinResolveResult::FailedWithError;
+	}
+
+	if (FlowPin->GetPinType() != PinType)
+	{
+		return EFlowDataPinResolveResult::FailedMismatchedType;
+	}
+
+	return EFlowDataPinResolveResult::Success;
+}
+
+// Must implement TryResolveDataPinAs...() for every EFlowPinType
+FLOW_ASSERT_ENUM_MAX(EFlowPinType, 12);
+
+template <typename TFlowDataPinResultType, EFlowPinType PinType>
+bool TResolveDataPinWorkingData<TFlowDataPinResultType, PinType>::TrySetupWorkingData(const FName& PinName, const UFlowNodeBase& FlowNodeBase)
+{
+	DataPinResult.Result = FlowNodeBase.TryResolveDataPinPrerequisites(PinName, FlowNode, FlowPin, PinType);
+	if (DataPinResult.Result != EFlowDataPinResolveResult::Success)
+	{
+		return false;
+	}
+
+	if (!FlowNode->TryGetFlowDataPinSupplierDatasForPinName(FlowPin->PinName, PinValueSupplierDatas))
+	{
+		return false;
+	}
+
+	// If we could not build the PinValueDataSuppliers array, 
+	// then the pin must be disconnected and have no default value available.
+	DataPinResult.Result = EFlowDataPinResolveResult::FailedUnconnected;
+
+	return true;
+}
+
+FFlowDataPinResult_Bool UFlowNodeBase::TryResolveDataPinAsBool(const FName& PinName) const
```

Comment:
The working data function simplifies these Resolve functions, but ultimately, we need one boilerplate for each pin type to interface with the data suppliers.